src/yunohost/certificate.py

```diff
@@ -0,0 +1,717 @@
+            raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_attempt_to_replace_valid_cert', domain=domain))
+
+
+        date_tag = datetime.now().strftime("%Y%m%d.%H%M%S")
+        new_cert_folder = "%s/%s-history/%s-selfsigned" % (CERT_FOLDER, domain, date_tag)
+
+        os.makedirs(new_cert_folder)
+
+        # Get serial
+        ssl_dir = '/usr/share/yunohost/yunohost-config/ssl/yunoCA'
+        with open(os.path.join(ssl_dir, 'serial'), 'r') as f:
+            serial = f.readline().rstrip()
+
+        shutil.copyfile(os.path.join(ssl_dir, "openssl.cnf"), os.path.join(new_cert_folder, "openssl.cnf"))
+
+        # FIXME : should refactor this to avoid so many os.system() calls...
+        # We should be able to do all this using OpenSSL.crypto and os/shutil
+        command_list = [
+            'sed -i "s/yunohost.org/%s/g" %s/openssl.cnf' % (domain, new_cert_folder),
+            'openssl req -new -config %s/openssl.cnf -days 3650 -out %s/certs/yunohost_csr.pem -keyout %s/certs/yunohost_key.pem -nodes -batch 2>/dev/null'
+            % (new_cert_folder, ssl_dir, ssl_dir),
+            'openssl ca -config %s/openssl.cnf -days 3650 -in %s/certs/yunohost_csr.pem -out %s/certs/yunohost_crt.pem -batch 2>/dev/null'
+            % (new_cert_folder, ssl_dir, ssl_dir),
+        ]
+
+        for command in command_list:
+            if os.system(command) != 0:
+                raise MoulinetteError(errno.EIO, m18n.n('certmanager_domain_cert_gen_failed'))
+
+        os.symlink('/etc/ssl/certs/ca-yunohost_crt.pem', os.path.join(new_cert_folder, "ca.pem"))
+        shutil.copyfile(os.path.join(ssl_dir, "certs", "yunohost_key.pem"), os.path.join(new_cert_folder, "key.pem"))
+        shutil.copyfile(os.path.join(ssl_dir, "newcerts", "%s.pem" % serial), os.path.join(new_cert_folder, "crt.pem"))
+
+        # append ca.pem at the end of crt.pem
+        with open(os.path.join(new_cert_folder, "ca.pem"), "r") as ca_pem:
+            with open(os.path.join(new_cert_folder, "crt.pem"), "a") as crt_pem:
+                crt_pem.write("\n")
+                crt_pem.write(ca_pem.read())
+
+        _set_permissions(new_cert_folder, "root", "root", 0755)
+        _set_permissions(os.path.join(new_cert_folder, "key.pem"), "root", "metronome", 0640)
+        _set_permissions(os.path.join(new_cert_folder, "crt.pem"), "root", "metronome", 0640)
+        _set_permissions(os.path.join(new_cert_folder, "openssl.cnf"), "root", "root", 0600)
+
+        _enable_certificate(domain, new_cert_folder)
+
+        # Check new status indicate a recently created self-signed certificate,
+        status = _get_status(domain)
+
+        if status and status["CA_type"]["code"] == "self-signed" and status["validity"] > 3648:
+            logger.success(m18n.n("certmanager_cert_install_success_selfsigned", domain=domain))
+        else :
+            logger.error("Installation of self-signed certificate installation for %s failed !", domain)
+            logger.error(str(e))
```

Comment:
That was a typo from copy-pasting, I removed the line. Unfortunately, I don't think there's much more info we can give to the user, since it's just a sanity check that the new current cert is a valid self-signed cert...